tests/support/server.rs

```diff
@@ -55,10 +56,16 @@ pub fn spawn(txns: Vec<Txn>) -> Server {
             }
 
             if let Some(dur) = txn.write_timeout {
-                let headers_end = ::std::str::from_utf8(&reply).unwrap().find("\r\n\r\n").unwrap() + 4;
+                let headers_end = unsafe { ::std::str::from_utf8_unchecked(&reply) }.find("\r\n\r\n").unwrap() + 4;
```

Comment:
Because gzipped content is not valid UTF8 and we'll always hit the sequence before the invalid content. Again, this is a test server so it's not so important that this is unsafe. If something unexpected happens then the test will fail.

But the road to ruin is paved with good intentions so I think it's reasonable to call this out and do it in an alternative way without the unsafe.